packages/labs/eleventy-plugin-lit/src/test/eleventy-plugin-test.ts

```diff
@@ -104,6 +104,53 @@ test.after.each(async ({rig}) => {
   await rig.cleanup();
 });
 
+/**
+ * Retry the given function until it returns a promise that resolves, or until
+ * the given timeout expires.
+ */
+const retryUntilTimeElapses = <T>(
+  timeout: number,
+  pollInterval: number,
+  fn: () => Promise<T>
+): Promise<T> => {
+  const start = performance.now();
+  let lastError: unknown;
+  return new Promise<T>((resolve, reject) => {
```

Comment:
Would an async function work? Might be more readable:

```ts
while (performance.now() - start < timeout) {
  try {
    return await fn();
  } catch (err) {
    lastError = err;
    await new Promise((r) => setTimeout(r, pollInterval));
  }
}
throw lastError ?? new Error('Timed out immediately'));
```

Reply:
Good call, done.